libs/designer/src/lib/core/state/workflow/workflowSelectors.ts

```diff
@@ -180,25 +180,50 @@ export const useNodeGraphId = (nodeId: string): string => {
   );
 };
 
-export const useGetAllAncestors = (nodeId: string) => {
+// BFS search for nodeId
+const getChildrenOfNodeId = (childrenNodes: Set<string>, rootNode?: WorkflowNode, nodeId?: string) => {
+  if (!rootNode) return undefined;
+
+  const queue = new Queue<WorkflowNode>();
+  queue.enqueue(rootNode);
+
+  while (queue.size > 0) {
+    const current = queue.dequeue();
+    if (current && current.id === nodeId) return getAllChildren(current, childrenNodes);
+    if (current?.id === nodeId) {
+      return current;
+    }
+
+    if (current?.children) {
+      for (const child of current.children) {
+        queue.enqueue(child);
+      }
+    }
+  }
+
+  return undefined;
+};
+
+// Adds all childrenIds
+const getAllChildren = (currNode: WorkflowNode, childrenNodes: Set<string>) => {
+  if (currNode.children) {
+    for (const child of currNode.children) {
+      getAllChildren(child, childrenNodes);
+    }
+  } else if (currNode.type === WORKFLOW_NODE_TYPES.OPERATION_NODE) {
```

Comment:
How will this work for case nodes?
So if switch is a children of some scope action, wont the children of switch be all subgraphnodes then how do we get for actions inside case nodes?

Reply:
This function is just returning the parent node if it's an operational node, and recursing to find all operation nodes within a subgraph node. Because switch node cases are based on the children, it does still calculate to get the operation nodes within a switch case